### PR TITLE
python3Packages.mkdocs-markdownextradata-plugin: init at 0.2.4

### PR DESCRIPTION
**Pull request by snpschaaf**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by snpschaaf** [comment]
Could we get this upstream @DavHau @hsjobeki ?

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6933,12 +6933,13 @@ self: super: with self; {
   mkdocs-autorefs = callPackage ../development/python-modules/mkdocs-autorefs { };
   mkdocs-drawio-exporter = callPackage ../development/python-modules/mkdocs-drawio-exporter { };
   mkdocs-exclude = callPackage ../development/python-modules/mkdocs-exclude { };
-  mkdocs-jupyter = callPackage ../development/python-modules/mkdocs-jupyter { };
-  mkdocs-gitlab = callPackage ../development/python-modules/mkdocs-gitlab-plugin { };
   mkdocs-git-authors-plugin = callPackage ../development/python-modules/mkdocs-git-authors-plugin { };
+  mkdocs-gitlab = callPackage ../development/python-modules/mkdocs-gitlab-plugin { };
   mkdocs-git-revision-date-localized-plugin = callPackage ../development/python-modules/mkdocs-git-revision-date-localized-plugin { };
+  mkdocs-jupyter = callPackage ../development/python-modules/mkdocs-jupyter { };
```
Comment (nikstur) [inline pkgs/top-level/python-packages.nix:6939]: Why are these lines changed?

File: pkgs/development/python-modules/mkdocs-markdownextradata-plugin/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+, mkdocs
+, pytestCheckHook
+, pythonOlder
+, pyyaml
+, click
+}:
+
+buildPythonPackage rec {
+  pname = "mkdocs-markdownextradata-plugin";
+  version = "0.2.4";
```
Comment (nikstur) [inline pkgs/development/python-modules/mkdocs-markdownextradata-plugin/default.nix:13]: This tools seems abandoned as it hasn't received any updates for 2 years. Are you sure we should include this in Nixpkgs?

File: pkgs/development/python-modules/mkdocs-markdownextradata-plugin/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  propagatedBuildInputs = [
+    mkdocs
+    pyyaml
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    click
+    pyyaml
```
Comment (nikstur) [inline pkgs/development/python-modules/mkdocs-markdownextradata-plugin/default.nix:33]: These don't seem to be necessary.

```suggestion
```

File: pkgs/development/python-modules/mkdocs-markdownextradata-plugin/default.nix (hunk 2 above)
Comment (snpschaaf) [reply pkgs/development/python-modules/mkdocs-markdownextradata-plugin/default.nix:13]: Your right! 

Going to use the packaged [`mkdocs-macros`](https://github.com/fralau/mkdocs_macros_plugin) plugin.

Closing this PR now.

**Comment by snpschaaf** [comment]
Closing as the package seem to be unmaintained. See https://github.com/NixOS/nixpkgs/pull/271193#discussion_r1430050039 for an alternative solution,

Closing this PR now.
